python/cudf/cudf/core/frame.py

```diff
@@ -2693,7 +2693,14 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        if self.iloc[0].dtype.kind in ("i", "u"):
+            data_type = cudf.utils.dtypes.min_column_type(self.astype('float')._column, None)
+            data = self.astype(data_type)
```

Comment:
@kkraus14 do you think this is okay as we'd be doing type-cast twice?

Reply:
What's the bigger picture goal of these two lines?

Reply:
We are essentially type-casting to the lowest possible float types(decide whether data can fit into `float32`/`float64`).

For that we are actually type-casting to `float64` and calling `min_column_type` to determine if it can be got down to `float32`. and then type-cast to `data_type`.

To avoid this double type-cast, My opinion is to build out an API to do something like this..

```python
def get_min_float_dtype(col):
    max_bound_dtype = np.min_scalar_type(float(x.max()))
    min_bound_dtype = np.min_scalar_type(float(x.min()))
    result_type = np.promote_types(max_bound_dtype, min_bound_dtype)
    if result_type == np.dtype("float16"):
    	# cuDF does not support float16 dtype
        result_type = np.dtype("float32")
    return result_type
```

Reply:
+1 to @galipremsagar's apparoach. Sounds good to me!

Reply:
Sounds good, I'll implement this instead!

Reply:
Should I add this to the dtypes.py file @galipremsagar?

Reply:
I've added the method to dtypes.py and updated my code in frame.py. It's working but the tests are failing, I think it's because pandas is returning Series/DF with float64 and we are now returning all float32's. Should I change the tests so they ignore the datatypes?